client/src/reducers/grantsMetadata.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  GrantMetadataActions,
+  GRANT_METADATA_LOADING_URI,
+  GRANT_METADATA_LOADING,
+  GRANT_METADATA_FETCHED,
+} from "../actions/grantsMetadata";
+import { Metadata } from "../types";
+
+export interface GrantsMetadataState {
+  [id: number]: {
+    loading: boolean;
+    metadata: Metadata | undefined;
+  };
+}
+
+export const initialState: GrantsMetadataState = {};
+
+export const grantsMetadataReducer = (
+  state: GrantsMetadataState = initialState,
+  action: GrantMetadataActions
+): GrantsMetadataState => {
+  switch (action.type) {
+    case GRANT_METADATA_LOADING_URI: {
```

Comment:
`GRANT_METADATA_LOADING` and this action type seem super similar. Can they be consolidated?

Reply:
@schultztimothy in case the metadata is already cached, we don't dispatch `GRANT_METADATA_LOADING`. 
the first idea was just to skip it, but I wanted another event before that, while we interact with the contract, to set the loading state. this also allow us to dispatch different errors if it fails to interact with the contract or later with ipfs